test/unit/LibPrices.js

```diff
@@ -317,12 +326,12 @@ describe("Unit tests: LibPrices.sol", function () {
                 // generate 24 hour oracle and tracer prices
                 for (i = 0; i < 24; i++) {
                     oraclePrices.push([
-                        ethers.utils.parseUnits((1 + 1 * i).toString(), 18),
-                        ethers.BigNumber.from("1"),
+                        ethers.utils.parseEther((1 + 1 * i).toString()),
+                        ethers.utils.parseEther("1"),
                     ])
                     tracerPrices.push([
-                        ethers.utils.parseUnits((1 + 0.5 * i).toString(), 18),
-                        ethers.BigNumber.from("1"),
```

Comment:
Could you add a test that has non-uniform or non-1 `trades` values?
There's the highlighted one, which just has `trades == 1`, and one above, which puts both `price` and `trades` to equal `i + 1` on each. Would be good to have a test covering just like a typical case where both are sort of random

Reply:
I'll add these in the `LibPricing` additional tests PR and resolve this once I have

Reply:
Nevermind, I did it in this PR; just have a look if that looks all good since it'd be good to get this merged in before the long awaited new functional tests